python/lsst/ts/standardscripts/scheduler/set_desired_state.py

```diff
@@ -0,0 +1,223 @@
+# This file is part of ts_standardscripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["SetDesiredState"]
+
+import types
+import typing
+import asyncio
+
+from lsst.ts import salobj
+from lsst.ts.idl.enums.Scheduler import SalIndex
+
+
+class SetDesiredState(salobj.BaseScript):
+    """A base script that implements setting the desired state for the
+    Scheduler.
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+    descr : `str`
+        Script description.
+    scheduler_index : `int`
+        Index of the Scheduler to enable.
```

Comment:
Thank you for adding the `desired_state` argument. Please document it here.

Also please consider generating the descr in the constructor.

Reply:
Please consider generating the descr in the constructor instead of taking it as an argument, e.g.:
```
# Cast the inputs to make sure they are valid
scheduler_index = SalIndex(scheduler_index)
desired_state = salobj.State(desired_state)
descr = f"Send {scheduler_index.name} to state {desired_state.name}"
```